apache/trafficcontrol
# TP: Ability to manager capabilities of servers

## What does this PR (Pull Request) do?
This PR provides users with the ability to:
- assign/remove server capabilities to/from a server
- view all servers that have a specific capability and, if needed, remove a server from a capability.

- [x] This PR is not related to any Issue but is related to this [blueprint](https://github.com/apache/trafficcontrol/blob/master/blueprints/server-capabilitites.md)

## Which Traffic Control components are affected by this PR?

- Traffic Portal

## What is the best way to verify this PR?

- Launch TP
- ensure that server capabilities exist: tp.domain.com/#!/server-capabilities (if not, add one or two)
- navigate to any server: tp.domain.com/#!/servers/{id}
- click on More > Manage Capabilities
- add/remove capabilities to/from the server but leave at least one capability on the server.
- navigate back to a capability you left on the server - tp.domain.com/#!/server-capabilities/{capability name}
- click 'view servers' to see the servers that utilize the capability and remove a server from the capability.

## The following criteria are ALL met by this PR


## Additional Information
Capabilities assigned to a server:

![image](https://user-images.githubusercontent.com/251272/66955497-b69e7080-f01f-11e9-8a20-b21d9b822079.png)

Servers that utilize a capability:

![image](https://user-images.githubusercontent.com/251272/67019296-acc94b80-f0b9-11e9-8afe-830a868b8eff.png)
